### ISSUE TITLE
Mismatched docs/implementation for getField

### ISSUE THREAD

**Issue by brunoseivam** [epics-base/pvDataCPP#66]
The documentation for `Structure::getField(size_t)` and `Union::getField(size_t)` claim that they return a null pointer if the index is not valid. The implementation throws a `out_of_range` exception instead (from [`vector::at`](http://www.cplusplus.com/reference/vector/vector/at/)). 

Also, `Union::getField(size_t)` returns `FieldConstPtr` but `Structure::getField(size_t)` returns `const FieldConstPtr &`. I think both should return `FieldConstPtr` for consistency.

`Structure`:

https://github.com/epics-base/pvDataCPP/blob/cd2436342d71b173eb4377e61d396e348c15546a/src/pv/pvIntrospect.h#L734-L740

`Union`:

https://github.com/epics-base/pvDataCPP/blob/cd2436342d71b173eb4377e61d396e348c15546a/src/pv/pvIntrospect.h#L857-L863

**Comment by brunoseivam** [comment]
Potentially fixed via #67

**Comment by mdavidsaver** [comment]
Looking back, it seems that `Structure::getField(size_t)` has never returned NULL.

It's original form in 2012 (87bff33c302a16ff4b4716fd6d5538a9e10af7c7) was

```c++
FieldConstPtr getField(std::size_t index) {return fields[index];}
```

And it is now:

```c++
const FieldConstPtr& getField(std::size_t index) const {return fields.at(index);}
```

So originally, out of bounds access was simply undefined (probably SIGSEGV).  Now it throws an exception.
